data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/config/CwlSinkConfig.java

```diff
@@ -0,0 +1,53 @@
+package org.opensearch.dataprepper.plugins.sink.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+
+public class CwlSinkConfig {
+    public static final String DEFAULT_BUFFER_TYPE = "in_memory";
+
+    //Class was utilized from the
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsConfig awsConfig;
+
+    @JsonProperty("threshold_config")
+    @NotNull
+    private ThresholdConfig thresholdConfig;
+
+    @JsonProperty("buffer_type")
+    private String bufferType = DEFAULT_BUFFER_TYPE;
```

Comment:
what are other options ?

Reply:
Currently there is only a default option for the buffer, and that is InMemoryBuffer, but this was made an option as in the future we could implement different buffering methods for customers who want to handle that application aspect differently. Similar to how the S3 handles InMemoryBuffering and InFileBuffering.

Reply:
Currently within the scope of the Design of this plugin, there are no goals for this. But I have added the addition of other Buffering systems such as Kafka and File Buffering as Next Step additions that could benefit the plugin for future usage. The generalization of this Buffer will prove useful as our services grow and customers require variable sources for which to buffer information. It would make it easier for downstream additions to be applied and integrated.